### PR TITLE
Fix bug 252

### PR DESCRIPTION
**Pull request by datnguyenhuy**
#252

### CODE REVIEW

File: lib/presentation/widget/upload_file/upload_file_widget.dart [hunk 1]
```diff
@@ -258,6 +266,13 @@ class _UploadFileWidgetState extends State<UploadFileWidget> {
     );
   }
 
+  String displayFileName(String filename)
```
Comment (huynguyennovem) [inline lib/presentation/widget/upload_file/upload_file_widget.dart:269]: consider to move this utility function into `StringExtension`

File: lib/presentation/widget/upload_file/upload_file_widget.dart [hunk 2]
```diff
@@ -557,3 +572,25 @@ class _UploadFileWidgetState extends State<UploadFileWidget> {
         });
   }
 }
+
+class MyTextPainter extends CustomPainter {
```
Comment (huynguyennovem) [inline lib/presentation/widget/upload_file/upload_file_widget.dart:576]: Let's change `MyTextPainter` into a more meaningful name. Such as: `Ellipsis TextPainter`

File: lib/presentation/widget/upload_file/upload_file_widget.dart [hunk 3]
```diff
@@ -258,6 +266,13 @@ class _UploadFileWidgetState extends State<UploadFileWidget> {
     );
   }
 
+  String displayFileName(String filename)
+  {
+    final sublength = filename.length - filename.split('.').last.length;
```
Comment (huynguyennovem) [inline lib/presentation/widget/upload_file/upload_file_widget.dart:271]: let's follow `lowerCamelCase` for variables: https://dart.dev/guides/language/effective-dart/style

File: lib/presentation/widget/upload_file/upload_file_widget.dart [hunk 4]
```diff
@@ -268,8 +268,8 @@ class _UploadFileWidgetState extends State<UploadFileWidget> {
 
   String displayFileName(String filename)
   {
-    var a = filename.length - filename.split(".").last.length;
-    return "..." + filename.substring(a-2,a) + filename.split('.').last;
+    final sublength = filename.length - filename.split('.').last.length;
+    return '...' + filename.substring(sublength-3,sublength) + filename.split('.').last;
```
Comment (hoangdat) [inline lib/presentation/widget/upload_file/upload_file_widget.dart:272]: Please create an extension method for that for easier in understanding

File: domain/lib/src/extension/string_extension.dart [hunk 5]
```diff
@@ -33,4 +33,10 @@ import 'package:domain/domain.dart';
 extension StringExtension on String {
   int compareToSort(String value, OrderType orderType) =>
       toLowerCase().compareTo(value.toLowerCase()) * (orderType == OrderType.ascending ? -1 : 1);
+
+  String displayFileName()
```
Comment (huynguyennovem) [inline domain/lib/src/extension/string_extension.dart:37]: - Need to reformat code to have better looking. 
- Check lint for coding convention. 
- And also set meaningful name for the function

For e.g:
```
  String formatFileNameEllipsis() {
    final subLength = length - split('.').last.length;
    return '...' + substring(subLength - 3, subLength) + split('.').last;
  }
```

Comment (hoangdat) [reply domain/lib/src/extension/string_extension.dart:37]: `toCenterEllipsis()`

File: lib/presentation/view/multiple_selection_bar/shared_space_multiple_selection_action_builder.dart [hunk 6]
```diff
@@ -11,7 +11,8 @@
 // subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
 // display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
 // the words “You are using the Free and Open Source version of LinShare™, powered by
-// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to
+// an Enterprise
```
Comment (hoangdat) [inline lib/presentation/view/multiple_selection_bar/shared_space_multiple_selection_action_builder.dart:15]: hmm

File: domain/lib/src/extension/string_extension.dart [hunk 7]
```diff
@@ -33,4 +33,10 @@ import 'package:domain/domain.dart';
 extension StringExtension on String {
   int compareToSort(String value, OrderType orderType) =>
       toLowerCase().compareTo(value.toLowerCase()) * (orderType == OrderType.ascending ? -1 : 1);
+
+  String toMiddleEllipsis()
+  {
+    final subLength = length - split('.').last.length;
+    return '...' + substring(subLength-3,subLength) + split('.').last;
+  }
```
Comment (hoangdat) [inline domain/lib/src/extension/string_extension.dart:41]: it is not C#. 
```dart
String toMiddleEllipsis() {
    final subLength = length - split('.').last.length;
    return '...' + substring(subLength-3,subLength) + split('.').last;
}
```
Please try to follow the convention in code base.

Comment (datnguyenhuy) [reply domain/lib/src/extension/string_extension.dart]: I fixed it and pushed it up. but still use -f. when not using -f has reported reject error. Hope you help me. Thank you, please

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Review by huynguyennovem — APPROVED** [review]

**Pull request merged**